azavea/raster-foundry
# Add ToolRun model

## Overview

We want a place to store configurations of runs of tools. This commit adds
a model for storing those parameters.

### Checklist


## Notes

Filtering, etc. tests will be added with the endpoint

## Testing Instructions

 * `./scripts/console app-server ./sbt`
 * `mg update`
 * `mg apply`
 * `app/console`
 * Run the following:
```scala
val user = Await.result(Users.getUserById("default"), 3 seconds).get
// Projects doesn't currently follow the pattern in other tables of inserting the `.Create` version
val project = Project.Create(publicOrgId, "Important EO Analysis Very Serious", "Learning everything about everything on EARTH", Visibility.Public, List.empty[String]).toProject(user.id)
val projectInserted = Await.result(Projects.insertProject(project), 3 seconds)
val tool = Await.result(database.db.run(Tools.take(1).result), 3 seconds)(0)
val toolRun = ToolRun.Create(Visibility.Public, publicOrgId, projectInserted.id, tool.id, Map("param" -> 3))
// should succeed:
Await.result(database.db.run(ToolRuns.insertToolRun(toolRun, user.id)), 3 seconds)
```
